### PR TITLE
Create show page

### PR DESCRIPTION
**Pull request by arpepper139**
We created our API endpoint for our show page, edited our FloofShowContainer to fetch from our endpoint, created a FloofShowTile to show the floof data, wrote the necessary tests (which were passing) and renamed our previous FloofTile component to FloofIndexTile to better distinguish between our Tiles.

### CODE REVIEW

File: app/javascript/containers/FloofShowContainer.js [hunk 1]
```diff
@@ -1,9 +1,46 @@
...
+      floof: {}
+    }
+  }
 
-  return(
-    <h1>Hello from Floof Show container</h1>
-  )}
+  componentDidMount() {
+    let id = this.props.params.id;
+    fetch(`/api/v1/floofs/${id}.json`)
```
Comment (dpickett) [inline app/javascript/containers/FloofShowContainer.js:14]: well done

**Review by dpickett — CHANGES_REQUESTED** [review]
I left some commentary and also spoke with part of your team about relying on the database for your categories

**Comment by arpepper139** [comment]
Commenting for self/team: We need to (1) create a "Categories" table, (2) and leverage that categories table to sort our floofs by category. We'll need to revisit our ER diagram for this. 

Also, we should give a unique ID to each array of categories we create per Brianna's comments.

**Comment by arpepper139** [comment]
We (1) created the categories table (2) leverages the categories table and serialization to return our Floofs by category, (3) refactored our React environment to utilize our new API data structures (for both the show and index page), and (4) refactored tests to account for changes. All tests were passing at commit.

File: app/serializers/category_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class CategorySerializer < ActiveModel::Serializer
+  attributes :id, :name, :associated_floofs
+
+  def associated_floofs
```
Comment (bkincart) [inline app/serializers/category_serializer.rb]: You can actually just do this with:
```
has_many :floofs
```

File: app/serializers/floof_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class FloofSerializer < ActiveModel::Serializer
+  attributes :id, :name, :job_title, :current_employer, :species, :category
+
+  def category
+    object.category.name
+  end
```
Comment (bkincart) [inline app/serializers/floof_serializer.rb]: Do you still need to serve up the category name this way? Are you using it somewhere? If not, I'd suggest removing this!

File: db/migrate/20180126143956_change_floofs_category_column.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    end
+  end
+
+  def down
+    change_table :floofs do |table|
+      table.remove_belongs_to :category
+      table.string :category, null: false
+    end
+  end
+end
```
Comment (bkincart) [inline db/migrate/20180126143956_change_floofs_category_column.rb:15]: Beautiful migration

**Review by bkincart — CHANGES_REQUESTED** [review]

File: app/javascript/containers/FloofIndexContainer.js [hunk 5]
```diff
@@ -34,13 +34,14 @@ class FloofIndexContainer extends Component {
     let floofCategoryTileComponents;
 
     if (floofData !== []) {
+      debugger
       floofCategoryTileComponents = floofData.map((category) => {
-        if (category.associated_floofs[0]) {
+        if (category.floofs[0]) {
```
Comment (bkincart) [inline app/javascript/containers/FloofIndexContainer.js]: I would almost prefer something a little more explicit here, such as `if(category.floofs !== [])`

**Pull request merged**